OrbitLinuxTracing/PerfEventProcessor.cpp

```diff
@@ -56,22 +56,21 @@ std::unique_ptr<PerfEvent> PerfEventQueue::PopEvent() {
 }
 
 void PerfEventProcessor::AddEvent(int origin_fd, std::unique_ptr<PerfEvent> event) {
-#ifndef NDEBUG
-  if (last_processed_timestamp_ > 0 &&
-      event->GetTimestamp() < last_processed_timestamp_ - kProcessingDelayMs * 1'000'000) {
-    ERROR("Processed an event out of order");
+  if (last_processed_timestamp_ns_ > 0 && event->GetTimestamp() < last_processed_timestamp_ns_) {
+    if (discarded_out_of_order_counter_ != nullptr) {
+      ++(*discarded_out_of_order_counter_);
+    }
+    return;
   }
-#endif
   event_queue_.PushEvent(origin_fd, std::move(event));
 }
 
 void PerfEventProcessor::ProcessAllEvents() {
   CHECK(!visitors_.empty());
   while (event_queue_.HasEvent()) {
     std::unique_ptr<PerfEvent> event = event_queue_.PopEvent();
-#ifndef NDEBUG
-    last_processed_timestamp_ = event->GetTimestamp();
-#endif
+    CHECK(event->GetTimestamp() >= last_processed_timestamp_ns_);
```

Comment:
I would add a comment why this CHECK must be true (events are discarded in AddEvent).